File: result/translate/starchat/python2java/java/FIND_PAIR_MAXIMUM_GCD_ARRAY.java
public static int f_gold ( int arr [ ], int n ) {
    int high = 0, i = 0;
    while ( i < n ) {
        high = Math.max ( high, arr [ i ] );
        i++;
    }
    int divisors [ ] = new int [ high + 1 ];
    i = 0;
    while ( i < n ) {
        int j = 1;
        while ( j <= (int) Math.sqrt ( arr [ i ] ) ) {
            if ( arr [ i ] % j == 0 ) {
                divisors [ j ] = divisors [ j ] + 1;
                if ( j!= arr [ i ] / j ) divisors [ arr [ i ] / j ] = divisors [ arr [ i ] / j ] + 1;
            }
            j++;
        }
        i++;
    }
    i = high;
    while ( i >= 1 ) {
        if ( divisors [ i ] > 1 ) return i;
        i--;
    }
    return 1;
}

File: result/translate/starchat/java2cpp/c++/FIND_PAIR_MAXIMUM_GCD_ARRAY.cpp
int f_gold ( int arr [ ], int n ) {
  int high = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) high = max ( high, arr [ i ] ) ;
  vector <int> divisors ( high + 1 ) ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    for ( int j = 1 ;
    j <= (int) sqrt ( arr [ i ] ) ;
    j ++ ) {
      if ( arr [ i ] % j == 0 ) {
        divisors [ j ] ++ ;
        if ( j!= arr [ i ] / j ) divisors [ arr [ i ] / j ] ++ ;
      }
    }
  }
  for ( int i = high ;
  i >= 1 ;
  i -- ) if ( divisors [ i ] > 1 ) return i ;
  return 1 ;
}